mlflow/sagemaker/__init__.py

```diff
@@ -1656,6 +1657,182 @@ def _does_model_exist(model_name, sage_client):
         return True if response else False
 
 
+class SageMakerPlugin(BaseDeploymentClient):
+    def __init__(self, target_uri):
+        super(SageMakerPlugin, self).__init__(target_uri=target_uri)
+
+    def _default_config(self):
+        return dict(
+            execution_role_arn=None,
+            bucket=None,
+            image_url=None,
+            region_name="us-west-2",
+            mode=DEPLOYMENT_MODE_CREATE,
+            archive=False,
+            instance_type=DEFAULT_SAGEMAKER_INSTANCE_TYPE,
+            instance_count=DEFAULT_SAGEMAKER_INSTANCE_COUNT,
+            vpc_config=None,
+            synchronous=True,
+            timeout_seconds=1200,
+        )
+
+    def _apply_custom_config(self, custom_config):
+        config = self._default_config()
+
+        for key in custom_config:
+            if key in config:
+                config[key] = custom_config[key]
+
+        return config
+
+    @experimental
+    def create_deployment(self, name, model_uri, flavor=None, config=None):
```

Comment:
Marking this as experimental because not all methods in this class have been implemented yet.